### DISCUSSION TITLE
Is it possible to await for refetches from `invalidateQueries`?

### DISCUSSION THREAD

**Discussion by trevordunn** [TanStack/query#4521] [Q&A] [answered]
The docs for `invalidateQueries` ([here](https://react-query-v3.tanstack.com/reference/QueryClient#queryclientinvalidatequeries)) say that "active queries are refetched in the background". Does that mean that when the promise resolves it's not guaranteed that the refetches are finished? If so, is there a way to wait for those refetches to finish?

If not, does that mean it's recommended to call `invalidateQueries` with `refetchActive` and `refetchInactive` set to `false`, then call `refetchQueries` directly after so you know when refetching is finished?

**Comment by TkDodo** [comment, marked as answer]
> Does that mean that when the promise resolves it's not guaranteed that the refetches are finished?

no, that's not what it means. If you choose to await the promise, the refetches are finished after that.

"background" means that we will not put the data into a hard loading state, but stale data will continue to be shown.

> If not, does that mean it's recommended to call invalidateQueries with refetchActive and refetchInactive set to false, then call refetchQueries directly after so you know when refetching is finished?

no, that's not necessary.

**Reply by trevordunn** [reply]
@TkDodo k. If I _did_ want to await the refetches, that's what I'd have to do though, correct?

**Reply by TkDodo** [reply]
if you want to avoid the default behaviour of invalidateQueries that refetches active queries, you have to pass `refetchType: 'none'`

**Reply by trevordunn** [reply]
Is `refetchType` a newly added filter? I don't see that anywhere in the docs or TS definitions.

**Reply by TkDodo** [reply]
yeah it's v4. In v3, you'd have to set `refetchActive: false` I think.
- https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4#query-filters

**Reply by marcospgp** [reply]
what about in v5?

**Reply by TkDodo** [reply]
v5 is the same as v4
